pydatalab/pydatalab/routes/v0_1/users.py

```diff
@@ -0,0 +1,34 @@
+from bson import ObjectId
+from flask import Blueprint, jsonify, request
+from flask_login import current_user
+
+from pydatalab.mongo import flask_mongo
+
+user = Blueprint("users", __name__)
+
+
+@user.route("/users/<user_id>", methods=["PATCH"])
+def save_user(user_id):
+    request_json = request.get_json()
+
+    display_name = request_json.get("display_name")
+    contact_email = request_json.get("contact_email")
+
+    if not current_user.is_authenticated:
+        return jsonify(status="error"), 401
+
+    if current_user.id != user_id and current_user.role != "admin":
+        return jsonify(status="error"), 403
+
+    update_result = flask_mongo.db.users.update_one(
+        {"_id": ObjectId(user_id)},
+        {"$set": {"display_name": display_name, "contact_email": contact_email}},
```

Comment:
Following the comments above, it could look something like this:

```suggestion
        {"$set": {"display_name": DisplayName(display_name), "contact_email": EmailStr(contact_email)}},
```

maybe with a try/except to give a clear error message in case of failure.